airflow/bin/cli.py

```diff
@@ -336,6 +336,47 @@ def list_tasks(args, dag=None):
         print("\n".join(sorted(tasks)))
 
 
+def parsedate_enhanced(args, dag):
+    '''
+    In addition to supporting date parsing, this also supports
+    @now,
+    @start_date (i.e. dag start date),
+    @end_date (i.e. dag end date if one exists), and
+    @latest (i.e. run closest to and before now that honors the schedule interval)
+    '''
+    timestr = args.execution_date
+    now = datetime.now()
+    start_date = dag.default_args['start_date'] if \
```

Comment:
I think default args is always a dict, so maybe just `start_date = dag.default_args.get('start_date')`? Same for end_date.